roseau/load_flow/tests/test_electrical_network.py

```diff
@@ -398,3 +450,111 @@ def test_to_from_dict_roundtrip(small_network: ElectricalNetwork):
     assert_frame_equal(small_network.branches_frame, new_net.branches_frame)
     assert_frame_equal(small_network.loads_frame, new_net.loads_frame)
     assert_frame_equal(small_network.voltage_sources_frame, new_net.voltage_sources_frame)
+
+
+def test_single_phase_network(single_phase_network: ElectricalNetwork):
+    # Test dict conversion
+    # ====================
+    net_dict = single_phase_network.to_dict()
+    new_net = ElectricalNetwork.from_dict(net_dict)
+    assert_frame_equal(single_phase_network.buses_frame, new_net.buses_frame)
+    assert_frame_equal(single_phase_network.branches_frame, new_net.branches_frame)
+    assert_frame_equal(single_phase_network.loads_frame, new_net.loads_frame)
+    assert_frame_equal(single_phase_network.voltage_sources_frame, new_net.voltage_sources_frame)
+
+    # Test load flow results
+    # ======================
+    source_bus = single_phase_network.buses["bus0"]
+    load_bus = single_phase_network.buses["bus1"]
+    line = single_phase_network.branches["line"]
+    load = single_phase_network.loads["load"]
+
+    json_results = {
+        "info": {
+            "status": "success",
+            "resolutionMethod": "newton",
+            "iterations": 1,
+            "targetError": 1e-06,
+            "finalError": 6.29e-14,
+        },
+        "buses": [
+            {"id": "bus0", "potentials": {"vb": [-10000.0, -17320.508], "vn": [0.0, 0.0]}},
+            {"id": "bus1", "potentials": {"vb": [-9999.974, -17320.464], "vn": [1.347e-12, 0.0]}},
+        ],
+        "branches": [
+            {
+                "id": "line",
+                "currents1": {"ib": [-0.0025, -0.0043], "in": [-1.347e-13, 0.0]},
+                "currents2": {"ib": [-0.0025, -0.0043], "in": [-1.347e-13, 0.0]},
+            }
+        ],
+        "loads": [
+            {"id": "load", "currents": {"ib": [-0.0025, -0.0043], "in": [-1.347e-13, 0.0]}},
+        ],
+        "sources": [],
+    }
+    solve_url = urljoin(ElectricalNetwork.DEFAULT_BASE_URL, "solve/")
+    with requests_mock.Mocker() as m:
+        m.post(solve_url, status_code=200, json=json_results, headers={"content-type": "application/json"})
+        single_phase_network.solve_load_flow(auth=("", ""))
+
+    # Test results of elements
+    # ------------------------
+    assert np.isclose(source_bus.potentials.m_as("V"), [-10000.0 - 17320.508j, 0j]).all()
+    assert np.isclose(load_bus.potentials.m_as("V"), [-9999.974 - 17320.464j, 1.347e-12 + 0j]).all()
+    assert np.isclose(line.currents[0].m_as("A"), [-0.0025 - 0.0043j, -1.347e-13 + 0j]).all()
+    assert np.isclose(line.currents[1].m_as("A"), [-0.0025 - 0.0043j, -1.347e-13 + 0j]).all()
+    assert np.isclose(load.currents.m_as("A"), [-0.0025 - 0.0043j, -1.347e-13 + 0j]).all()
```

Comment:
if you want, you can use `npt.assert_all_close(array1, array2)` with `import numpy.testing as npt`